client/state/current-user/gravatar-status/actions.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import request from 'superagent';
+
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE
+} from 'state/action-types';
+import {
+	GRAVATAR_CACHE_EXPIRATION
+} from 'state/current-user/gravatar-status/constants';
+
+export function uploadGravatar( file, bearerToken, email ) {
+	return dispatch => {
+		dispatch( { type: GRAVATAR_UPLOAD_REQUEST } );
+		const data = new window.FormData();
+		data.append( 'filedata', file );
+		data.append( 'account', email );
+		return request
+			.post( 'https://api.gravatar.com/v1/upload-image' )
+			.send( data )
+			.set( 'Authorization', 'Bearer ' + bearerToken )
+			.then( () => {
+				const fileReader = new window.FileReader( file );
+				fileReader.addEventListener( 'load', function() {
+					dispatch( {
+						type: GRAVATAR_UPLOAD_RECEIVE,
+						expiration: Date.now() + GRAVATAR_CACHE_EXPIRATION,
+						tempImage: fileReader.result
+					} );
+					dispatch( {
+						type: GRAVATAR_UPLOAD_REQUEST_SUCCESS
+					} );
+				} );
+				fileReader.readAsDataURL( file );
```

Comment:
I think we might want to add an onerror handler too, the error info should then be available off of https://developer.mozilla.org/en-US/docs/Web/API/FileReader/error